tests/core/pyspec/eth2spec/test/eip7732/epoch_processing/test_process_builder_pending_payments.py

```diff
@@ -0,0 +1,281 @@
+        (2, spec.MIN_ACTIVATION_BALANCE, quorum + 100),  # Above threshold
+        (3, spec.MIN_ACTIVATION_BALANCE, quorum - 1),  # Below threshold
+    ]
+
+    for i, (builder_index, amount, weight) in enumerate(payments_data):
+        if i >= spec.SLOTS_PER_EPOCH:
+            break
+        payment = create_builder_pending_payment(spec, builder_index, amount, weight)
+        state.builder_pending_payments[i] = payment
+
+    pre_builder_pending_withdrawals = len(state.builder_pending_withdrawals)
+
+    yield from run_epoch_processing_with(spec, state, "process_builder_pending_payments")
+
+    # Only payments with weight > quorum should be processed (indices 0 and 2)
+    expected_processed = 2
+    assert (
+        len(state.builder_pending_withdrawals)
+        == pre_builder_pending_withdrawals + expected_processed
+    )
+
+    # Check the processed withdrawals
+    processed_withdrawals = state.builder_pending_withdrawals[pre_builder_pending_withdrawals:]
+    processed_builder_indices = [w.builder_index for w in processed_withdrawals]
+    assert 0 in processed_builder_indices
+    assert 2 in processed_builder_indices
+    assert 1 not in processed_builder_indices
+    assert 3 not in processed_builder_indices
+
+
+@with_eip7732_and_later
+@spec_state_test
+def test_process_builder_pending_payments_queue_rotation(spec, state):
+    """Test that the payment queue is properly rotated after processing."""
+    # Advance past genesis epochs
+    next_epoch(spec, state)
+    next_epoch(spec, state)
+
+    # Fill both epochs of the queue with test data
+    test_weight = 12345
+    for i in range(2 * spec.SLOTS_PER_EPOCH):
+        payment = create_builder_pending_payment(spec, i, spec.MIN_ACTIVATION_BALANCE, test_weight)
+        state.builder_pending_payments[i] = payment
+
+    # Store the second epoch data for comparison
+    second_epoch_payments = []
+    for i in range(spec.SLOTS_PER_EPOCH, 2 * spec.SLOTS_PER_EPOCH):
+        payment = state.builder_pending_payments[i]
+        second_epoch_payments.append((payment.weight, payment.withdrawal.builder_index))
+
+    yield from run_epoch_processing_with(spec, state, "process_builder_pending_payments")
+
+    # First SLOTS_PER_EPOCH should be the old second epoch
+    for i in range(spec.SLOTS_PER_EPOCH):
+        payment = state.builder_pending_payments[i]
+        expected_weight, expected_builder_index = second_epoch_payments[i]
+        assert payment.weight == expected_weight
+        assert payment.withdrawal.builder_index == expected_builder_index
+
+    # Second SLOTS_PER_EPOCH should be empty
+    for i in range(spec.SLOTS_PER_EPOCH, 2 * spec.SLOTS_PER_EPOCH):
+        payment = state.builder_pending_payments[i]
+        assert payment.weight == 0
+        assert payment.withdrawal.amount == 0
+        assert payment.withdrawal.builder_index == 0
+
+
+@with_eip7732_and_later
+@spec_state_test
+def test_process_builder_pending_payments_large_amount_churn_impact(spec, state):
```

Comment:
Love this test! started the review already writing that this test will be missing, happy to cross this from my list :heart: